src/gui/guiKeyChangeMenu.cpp

```diff
@@ -153,9 +153,8 @@ void GUIKeyChangeMenu::regenerateGui(v2u32 screensize)
 		{
 			core::rect < s32 > rect(0, 0, 100, 30);
 			rect += topleft + v2s32(offset.X + 120, offset.Y - 5);
-			const wchar_t *text = wgettext(k->key.name());
-			k->button = Environment->addButton(rect, this, k->id, text);
-			delete[] text;
+			k->button = Environment->addButton(rect, this, k->id,
+							   k->key.name()[0] ? wgettext(k->key.name()) : L"");
```

Comment:
How about deleting the pointer from wgettext? How about this:
```C++
const wchar_t *text = wgettext(k->key.name());
k->button = Environment->addButton(rect, this, k->id,
	k->key.name()[0] ? text : L"");
delete[] text;
```

Reply:
`L""` is a static string in that case, you can't delete it

Reply:
I tried it yesterday with L"" and crash wen key change menu opens:

guadalinex@guadalinex:~/minetest$ bin/minetest
2018-03-27 20:36:09: ERROR[Main]: Subgame specified in default_game [minetest] is invalid.
Loaded texture: /home/guadalinex/minetest/games/minimal/menu/icon.png
Loaded texture: /home/guadalinex/minetest/textures/base/pack/menu_header.png
*** Error in `bin/minetest': munmap_chunk(): invalid pointer: 0x0852f2b0 ***
Abortado

Whit *L""  works in my linux but not in other systems, can't compile, so the only option is with an "if/else" or suppresing the "text" variable

Reply:
Thanks @SmallJoker you were right. I had to delete the pointer returned by wgettext as indicated in src/gettext.h